src/names.jl

```diff
@@ -152,3 +152,43 @@ function named_axiskeys(A::AbstractArray)
     NT = NamedTuple{dimnames(A)}
     NT(axiskeys(A))
 end
+
+
+"""
+    rekey(A, (1:10, [:a, :b]))
+    rekey(A, 2 => [:a, :b])
+    rekey(A, :y => [:a, :b])
+    rekey(A, :y => q => [:a, :b])
+
+Rekey a KeyedArray via `Tuple`s, `dim => newkey`. If `A` also has named dimensions then you
+can also pass `dimname => newkey`, or even `oldname => newname => newkey` to both `rename`
+and `rekey` the specified dimension.
+"""
+rekey(x::AbstractArray, k2::Tuple) = KeyedArray(keyless(x), k2)
```

Comment:
I wonder if this method ought to be restricted to types which have axis keys? 

Mostly because it seems a little odd verbally, and redundant to provide a constructor. But, on the other hand, maybe it's useful not to have to care?

Precedent:
```julia
julia> rename(rand(3), (:x,))
ERROR: MethodError: no method matching rename(::Vector{Float64}, ::Tuple{Symbol})
Closest candidates are:
  rename(::NamedDimsArray, ::Any) at /Users/me/.julia/dev/NamedDims/src/name_operations.jl:14
```

Reply:
Yeah, I mostly did this to avoid needing the awkward `rekey(A::Union{NdaKa{L,T,N}, KaNda{L,T,N}}, k2::Tuple) where {L, T, N}` syntax. I assume there isn't a better solution? I suppose if this code is being used more than read then it doesn't really matter, but it's really hard for my brain to parse that `Union` type.

Reply:
Could be just `A::Union{KeyedArray, NdaKa}` I think?

Reply:
Unless you wanted to split things so as to preserve the order of wrappers, with a separate `A:: NdaKa` method which unwraps the names & puts them back on afterwards.